docs/development/experimental_features.md

```diff
@@ -0,0 +1,37 @@
+# Implementing experimental features in Synapse
+
+It can be desirable to implement "experimental" features which are disabled by
+default and must be explicitly enabled via the Synapse configuration. This is
+applicable for features which:
+
+* Are unstable in the Matrix spec (e.g. those defined by an MSC).
+* Developers are not confident are ready for use by general Synapse administrators/users
+  (e.g. a feature is incomplete, buggy, performs poorly, or needs further testing).
+
+Note that this does *not* include features which are not expected to be desirable
+to a broad audience. The [module infrastructure](../modules/index.md) should be
+investigated for non-standard features.
```

Comment:
Is the suggestion here that some module features should be hidden behind experimental config flags, or?

Reply:
The suggestion is that some features should be built as modules, not as things behind experimental config flags.

Reply:
Maybe it is clearer to just remove this? Not sure, was trying to give people other options though.

Reply:
Aha OK! In that case, I might just slightly reword this to:

```suggestion
to a broad audience. The [module infrastructure](../modules/index.md) should
instead be investigated for non-standard features.
```